haystack/document_store/milvus.py

```diff
@@ -0,0 +1,384 @@
+        :param index: Index name for text, embedding and metadata.
+        :param vector_dim: The embedding vector size by default it use 768 dimension.
+        :param index_file_size: File size for Milvus server embedding vector store by default it use 1024 MB.
+        :param metric_type: Embedding vector search metrics by default it use IP.
+        :param index_type: Embedding vector indexing type by default it use FLAT.
+        :param index_param: Embedding vector index creation parameter by default it use {"nlist": 16384}.
+                            Refer for more information https://github.com/milvus-io/pymilvus/blob/master/doc/source/param.rst
+        :param search_param: Embedding vector search parameter by default it use {"nprobe": 10}.
+                             Refer for more information https://github.com/milvus-io/pymilvus/blob/master/doc/source/param.rst
+        :param update_existing_documents: Whether to update any existing documents with the same ID when adding
+                                          documents. When set as True, any document with an existing ID gets updated.
+                                          If set to False, an error is raised if the document ID of the document being
+                                          added already exists.
+        :param return_embedding: To return document embedding.
+        :param embedding_field: Name of field containing an embedding vector.
+        """
+        self.milvus_server = Milvus(uri=server_uri, pool=connection_pool)
+        self.vector_dim = vector_dim
+        self.index_file_size = index_file_size
+        self.metric_type = metric_type
+        self.index_type = index_type
+        self.index_param = index_param or {"nlist": 16384}
+        self.search_param = search_param or {"nprobe": 10}
+        self.index = index
+        self._create_collection_and_index_if_not_exist(self.index)
+        self.return_embedding = return_embedding
+        self.embedding_field = embedding_field
+
+        super().__init__(
+            url=sql_url,
+            update_existing_documents=update_existing_documents,
+            index=index
+        )
+
+    def __del__(self):
+        return self.milvus_server.close()
+
+    def _create_collection_and_index_if_not_exist(
+        self,
+        index: Optional[str] = None,
+        index_param: Optional[Dict[str, Any]] = None
+    ):
+        index = index or self.index
+        index_param = index_param or self.index_param
+
+        status, ok = self.milvus_server.has_collection(collection_name=index)
+        if not ok:
+            collection_param = {
+                'collection_name': index,
+                'dimension': self.vector_dim,
+                'index_file_size': self.index_file_size,
+                'metric_type': self.metric_type
+            }
+
+            self.milvus_server.create_collection(collection_param)
```

Comment:
Let's check the returned `status` of `create_collection()` and raise an exception if it is not `ok` (same for  `create_index()`). Otherwise, we might only learn about the failed init when starting to write docs there ...